websites/C/Club Penguin Rewritten/presence.ts

```diff
@@ -381,7 +381,76 @@ presence.on("UpdateData", async () => {
+          let n = "";
+          if (s == "videos") {
+            n = "Videos";
+          } else if (s == "news") {
+            n = "News";
+          } else if (s == "sneak-peeks") {
+            n = "Sneak Peeks";
+          } else if (s == "reviewed-by-you") {
+            n = "Reviewed by You";
+          } else if (s == "potw") {
+            n = "Penguin of the Week";
+          } presenceData.state = 'Filtering for "' + n + '"';
+        }
+      } else {
+        presenceData.largeImageKey = "newspaper";
+        presenceData.details = "Reading a Community Post";
+        presenceData.state = document.querySelector(".blog_head h4").textContent + " (" + document.querySelector(".blog_head").textContent.split(" |")[0].replace(document.querySelector(".blog_head h4").textContent, "").replace("\n","").replace("\n","") + ")";
+      }
+    } else if (document.location.host == "cprewritten.net") {
+      if (document.location.pathname == "/") {
+        presenceData.details = "Viewing Homepage";
+      } else if (document.location.pathname.includes("/help/")) {
+        presenceData.details = "Getting Help";
+        if (document.location.pathname == "/help/articles/your_penguin.html") {
+          presenceData.state = "Your Penguin";
+        } else if (document.location.pathname == "/help/articles/mini_games.html") {
+          presenceData.state = "Mini Games";
+        } else if (document.location.pathname == "/help/articles/navigation.html") {
+          presenceData.state = "Navigation";
+        } else if (document.location.pathname == "/help/articles/your_igloo.html") {
+          presenceData.state = "Your Igloo";
+        } else if (document.location.pathname == "/help/articles/penguin_coins.html") {
+          presenceData.state = "Penguin Coins";
+        } else if (document.location.pathname == "/help/articles/newspaper.html") {
+          presenceData.state = "Newspaper";
+        } else if (document.location.pathname == "/help/articles/characters.html") {
+          presenceData.state = "Characters";
+        } else if (document.location.pathname == "/help/articles/special_roles.html") {
+          presenceData.state = "Special Roles";
+        } else if (document.location.pathname == "/help/articles/secrets.html") {
+          presenceData.state = "Secrets";
+        } else if (document.location.pathname == "/help/articles/safety.html") {
+          presenceData.state = "Safety";
+        } else if (document.location.pathname == "/help/articles/other_penguins.html") {
+          presenceData.state = "Other Penguins";
+        } else if (document.location.pathname == "/help/articles/puffles.html") {
+          presenceData.state = "Puffles";
+        } else if (document.location.pathname == "/help/articles/communicating.html") {
+          presenceData.state = "Communicating";
+        } else if (document.location.pathname == "/help/articles/tutorial.html") {
+          presenceData.state = "Tutorial";
+        }
```

Comment:
Just a suggestion but I would recommend using switch case statements

Reply:
I'll read up on that.

Reply:
how's this? `else if (document.location.host == "cprewritten.net") {
      if (document.location.pathname == "/") {
        presenceData.details = "Viewing Homepage";
      } else if (document.location.pathname.includes("/help/")) {
        presenceData.details = "Getting Help";
        if (document.location.pathname.includes(".html")) {
          const uncaptializedState = document.location.pathname.replace("/help/articles/", "").replace(".html", "");
          let state = "";
          const words = uncaptializedState.split("_");
          for (let i = 0; i < words.length; i++) {
            if (i == 0) {
              state = words[i].charAt(0).toUpperCase() + words[i].slice(1);
            } else {
              state = state + " " + words[i].charAt(0).toUpperCase() + words[i].slice(1);
            }
          } presenceData.state = state;
        }
      }
    }`